easybuild/tools/job/pyslurm_backend.py

```diff
@@ -0,0 +1,146 @@
+        version = pyslurm_version()
+        if LooseVersion(version) < LooseVersion(self.REQ_VERSION):
+            raise EasyBuildError("Found pyslurm version %s, but version %s or more recent is required",
+                                 version, self.REQ_VERSION)
+
+    def init(self):
+        """
+        Initialise the job backend.
+
+        Connect to the PBS server & reset list of submitted jobs.
+        """
+        self._submitted = []
+
+    def queue(self, job, dependencies=frozenset()):
+        """
+        Add a job to the queue.
+
+        :param dependencies: jobs on which this job depends.
+        """
+        if dependencies:
+            job.job_specs['dependency'] = 'afterany:%s' % ':'.join(str(d.jobid) for d in dependencies)
+
+        # submit job with hold in place
+        job.job_specs['hold'] = True
+
+        self.log.info("Submitting job with following specs: %s", job.job_specs)
+        job.jobid = pyslurm.job().submit_batch_job(job.job_specs)
+
+        self._submitted.append(job)
+
+    def complete(self):
+        """
+        Complete a bulk job submission.
+
+        Release all user holds on submitted jobs, and disconnect from server.
+        """
+        for job in self._submitted:
+            if job.job_specs['hold']:
+                self.log.info("releasing user hold on job %s" % job.jobid)
+                run_cmd("scontrol release %s" % job.jobid)
+
+        submitted_jobs = '; '.join(["%s (%s): %s" % (job.name, job.module, job.jobid) for job in self._submitted])
+        print_msg("List of submitted jobs (%d): %s" % (len(self._submitted), submitted_jobs), log=self.log)
+
+    def make_job(self, script, name, env_vars=None, hours=None, cores=None):
+        """Create and return a job dict with the given parameters."""
+        return SlurmJob(script, name, env_vars=env_vars, hours=hours, cores=cores)
+
+
+class SlurmJob(object):
+    """Job class for SLURM jobs."""
+
+    def __init__(self, script, name, env_vars=None, hours=None, cores=None):
+        """Create a new Job to be submitted to SLURM."""
+        self.log = fancylogger.getLogger(self.__class__.__name__, fname=False)
+
+        self.jobid = None
+        self.script = script
+        self.name = name
+
+        self.job_specs = {'wrap': self.script, 'job_name': self.name}
+
+        if env_vars:
+            self.job_specs['export'] = ','.join(sorted(env_vars.keys()))
+
+        max_walltime = build_option('job_max_walltime')
+        if hours is None:
+            hours = max_walltime
+        if hours > max_walltime:
+            self.log.warn("Specified %s hours, but this is impossible. (resetting to %s hours)" % (hours, max_walltime))
+            hours = max_walltime
+        self.job_specs['time'] = '%s:0:0' % hours
+
+        if cores:
+            self.job_specs['nodes'] = '1'
+            # value passed here must be an integer!
+            self.job_specs['ntasks_per_node'] = cores
```

Comment:
And it must also be >0, should perhaps check user input here and fail nicely if it's bogus.

Reply:
This is already done in `create_job` in `tools/parallelbuild.py`, where we only pass down `cores` if it's non-zero.

Reply:
Ok.